src/EnergyPlus/FanCoilUnits.cc

```diff
@@ -5910,6 +5954,36 @@ namespace FanCoilUnits {
         return Residuum;
     }
 
+    void GetControlledZoneNum(EnergyPlusData &state, int const FanCoilNum, bool const NodeCheck)
+    {
+        auto &FanCoil(state.dataFanCoilUnits->FanCoil);
+        int ControlZoneNum = 0;
+
+        for (int CtrlZone = 1; CtrlZone <= state.dataGlobal->NumOfZones; ++CtrlZone) {
+            if (!state.dataZoneEquip->ZoneEquipConfig(CtrlZone).IsControlled) continue;
+            if (NodeCheck) {
+                for (int NodeNum = 1; NodeNum <= state.dataZoneEquip->ZoneEquipConfig(CtrlZone).NumInletNodes; ++NodeNum) {
+                    if (FanCoil(FanCoilNum).AirOutNode == state.dataZoneEquip->ZoneEquipConfig(CtrlZone).InletNode(NodeNum)) {
+                        FanCoil(FanCoilNum).NodeNumOfControlledZone = state.dataZoneEquip->ZoneEquipConfig(CtrlZone).ZoneNode;
+                        ControlZoneNum = CtrlZone;
+                        break;
+                    }
+                }
+            } else {
+                for (int Num = 1; Num <= state.dataZoneEquip->ZoneEquipList(CtrlZone).NumOfEquipTypes; ++Num) {
+                    if (UtilityRoutines::SameString(FanCoil(FanCoilNum).Name, state.dataZoneEquip->ZoneEquipList(CtrlZone).EquipName(Num)) &&
+                        UtilityRoutines::SameString(FanCoil(FanCoilNum).UnitType, state.dataZoneEquip->ZoneEquipList(CtrlZone).EquipType(Num))) {
+                        ControlZoneNum = CtrlZone;
+                        break;
+                    }
+                }
+            }
+            if (ControlZoneNum > 0) break;
+        }
+
+        FanCoil(FanCoilNum).ControlZoneNum = ControlZoneNum;
+    }
+
```

Comment:
There are two places to find zone number. I consolidate two places as a common function based on search scenario. The other cases may not need to use the zone number.